tests/fs.rs

```diff
@@ -535,18 +535,18 @@ fn mkdir_path_already_exists_error() {
     assert_eq!(e.kind(), ErrorKind::AlreadyExists);
 }
 
-/*
 #[test]
 fn recursive_mkdir() {
     let tmpdir = tmpdir();
     let dir = "d1/d2";
     check!(tmpdir.create_dir_all(dir));
-    assert!(dir.is_dir())
+    assert!(tmpdir.is_dir("d1"));
+    let dir = check!(tmpdir.open_dir("d1"));
+    assert!(dir.is_dir("d2"));
+    assert!(tmpdir.is_dir("d1/d2"));
```

Comment:
These extra checks are good, though would it be inconvenient to add them in a separate file? That way we can keep `fs.rs` as close to a direct translation of Rust's `fs.rs` as possible.